enhancements/network/drop-node-hostname-resolution.md

```diff
@@ -0,0 +1,198 @@
+  designs will need to take that into account. We also have on-prem ci jobs
+  running that would quickly catch any such problems.
+
+* A deployer may have third-party services in their cluster that depend on
+  node name resolution. It's debatable whether our internal DNS configuration
+  should be considered a public interface, and in such use cases the name
+  resolution could be provided by external DNS. Some baremetal IPI deployments
+  are already doing that.
+
+## Design Details
+
+### Open Questions [optional]
+
+No one I have talked to is aware of any current requirement for node hostname
+resolution, but no one has been able to say that there definitely isn't one
+either. Is there anyone who could give us a definitive answer on this?
+
+### Test Plan
+
+As this would become the default behavior for on-prem IPI deployments, it
+would be covered by existing tests. The full test suite should continue to
+pass after this change is made. This introduces no new behavior that would
+need to be tested.
+
+### Graduation Criteria
+
+This is not strictly a feature and wouldn't go through a graduation process.
+It's more of an RFC to confirm that we haven't missed anything. Once node
+name resolution is removed this will become the supported configuration for
+on-prem IPI deployments. Ideally, the change should be invisible to users,
+so there is no need for a dev or tech preview phase.
+
+#### Dev Preview -> Tech Preview
+
+N/A
+
+#### Tech Preview -> GA
+
+N/A
+
+#### Removing a deprecated feature
+
+N/A
+
+### Upgrade / Downgrade Strategy
+
+On upgrade, the coredns static pod will be removed and replaced with an
+instance of dnsmasq running on the host that will only provide the api-int
+record. We will need to resolve port conflicts between coredns and dnsmasq
+via either ordering or retries. As part of that, we will need to ensure that
+when the coredns pod is deleted it doesn't break the upgrade in such a way
+that the dnsmasq service cannot start and take over DNS responsibilities.
+
+On downgrade, the coredns pod would be recreated. It may be necessary to
+manually stop dnsmasq to allow coredns to start. I am not aware of any
+functionality in machine-config-operator that would allow us to automatically
+stop a service on downgrade.
+
+### Version Skew Strategy
+
+This component does not interact with anything outside of the node it is
+running on. If separate nodes end up at different levels, the only difference
+will be what is providing internal DNS to a given node. It should not be a
+problem to have one node at one version and another at a different version.
+
+## Implementation History
+
+## Drawbacks
+
+As noted earlier, removing node hostname resolution from on-prem platforms
+will introduce a difference in behavior from the cloud platforms. Since
+efforts have been made to avoid reliance on DNS I don't see this as a major
+problem, but it is less than ideal.
+
+## Alternatives
```

Comment:
Q: is the main motivation is remote workers case? 

If the answer to the Q is yes, do you think that the following could be an alternative (though this will compromise DNS performance) ?
1. Master nodes will continue running CoreDNS as today
2. Remove core-DNS pod from workers 
3. Update resolv.conf prepender script  on workers to point to one of the masters  (api-vip )